ui/src/store/operations.js

```diff
@@ -49,26 +59,24 @@ export function recentOrders(state, orders) {
   return { ...state, recentOrders: orders, orderbook, orderhistory };
 }
 
-export function createOffer(state, { isBuy, assetAmount, assetPurse, priceAmount, pricePurse }) {
-  const { instanceId } = state;
+export function createOffer(
+  state,
+  { isBuy, assetAmount, assetPurse, priceAmount, pricePurse },
+) {
+  const {
+    instanceHandleBoardId,
+    installationHandleBoardId,
+    inviteDepositId,
+  } = state;
   const now = Date.now();
   const offer = {
     // JSONable ID for this offer.  This is scoped to the origin.
     id: now,
 
-    // Contract-specific metadata.
-    instanceRegKey: instanceId,
-    contractIssuerIndexToKeyword: ['Asset', 'Price'],
-
-    // Format is:
-    //   hooks[targetName][hookName] = [hookMethod, ...hookArgs].
-    // Then is called within the wallet as:
-    //   E(target)[hookMethod](...hookArgs)
-    hooks: {
-      publicAPI: {
-        getInvite: ['makeInvite'], // E(publicAPI).makeInvite()
-      },
-    },
+    // TODO: get this from the invite instead in the wallet. We
```

Comment:
Not strictly necessary.  The UI is already trusting the dapp backend.

Reply:
Oh, you mean the wallet is trusting the backend? I think we can do better than that, though. If the wallet extracts the instanceHandle and installationHandle from the invite, then the wallet will be trusting Zoe instead

Reply:
This code is in the dapp ui, so I meant the ui is trusting the backend.

Reply:
Oh, yes, the UI is trusting the dapp backend. I think what I'm talking about here is that the wallet currently takes the instanceHandle and installationHandle unquestioningly from the dapp, but it shouldn't